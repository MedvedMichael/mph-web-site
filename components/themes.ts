const light = {
    bg: {
        primary: '#c8d8e3',
        secondary: '#ffffff',
        inset: '#e2e4e8',
        input: 'rgba(65,67,78,0.12)',
        nav: '#5785ba'
    },
    text: {
        primary: '#fbfbfc',
        secondary: '#2f3037',
        tertiary: '#525560',
        quarternary: '#9194a1',
        placeholder: 'rgba(82,85,96,0.5)',
        onPrimary: '#ffffff'
    },
    border: {
        nav: '1px solid #282828'
    }
    // ...
}

const dark = {
    bg: {
        primary: '#162333',
        secondary: '#111111',
        inset: '#111111',
        input: 'rgba(191,193,201,0.12)',
        nav: '#18304d'
    },
    text: {
        primary: '#fbfbfc',
        secondary: '#e3e4e8',
        tertiary: '#a9abb6',
        quarternary: '#6c6f7e',
        placeholder: 'rgba(145,148,161,0.5)',
        onPrimary: '#050505'
    },
    border: {
        nav: '1px solid #282828'
    }
    // ...
}

const defaultTheme = {
    fontSizes: [
        '14px', // 0
        '16px', // 1
        '18px', // 2
        '22px', // 3
        '26px', // 4
        '32px', // 5
        '40px'  // 6
    ],
    fontWeights: {
        body: 400,
        subheading: 500,
        link: 600,
        bold: 700,
        heading: 800,
    },
    lineHeights: {
        body: 1.5,
        heading: 1.3,
        code: 1.6,
    },
    // ...
};

export const lightTheme = { ...defaultTheme, ...light }
export const darkTheme = { ...defaultTheme, ...dark }