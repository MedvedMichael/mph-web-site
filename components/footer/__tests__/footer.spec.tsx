import { shallow } from 'enzyme'
import FooterView, {Footer, FooterGroup} from '../footer'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { lightTheme } from '../../themes'
import 'jest-styled-components'
import toJson from 'enzyme-to-json'

test ('should render', () => {
    const res = shallow(<FooterView/>)
    expect(toJson(res)).toMatchSnapshot()
})

test('should footer have correct styles', () => {
    const res = renderer.create(<ThemeProvider theme={lightTheme}><Footer/></ThemeProvider>).toJSON()
    expect(res).toHaveStyleRule('background', lightTheme.bg.nav)
    expect(res).toHaveStyleRule('transition', lightTheme.transition.bg)
    expect(res).toHaveStyleRule('color', lightTheme.text.primary)
})

test('should footerGroup have correct styles', () => {
    const res = renderer.create(<ThemeProvider theme={lightTheme}><FooterGroup/></ThemeProvider>).toJSON()
    expect(res).toHaveStyleRule('color', lightTheme.text.primary)
})