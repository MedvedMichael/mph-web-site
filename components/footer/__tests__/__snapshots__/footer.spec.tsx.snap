// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<styled.footer>
  <styled.div>
    <styled.div>
      <styled.div>
        <styled.h3>
          Location
        </styled.h3>
        <h5>
          Kiev, Ukraine
        </h5>
      </styled.div>
      <styled.div>
        <h3>
          Contact me
        </h3>
        <styled.div>
          <a
            href="https://t.me/medved2001"
            target="_blank"
          >
            <styled.i
              aria-hidden={true}
              className="fab fa-telegram-plane"
            />
          </a>
          <a
            href="https://www.facebook.com/profile.php?id=100006384134639"
            target="_blank"
          >
            <styled.i
              aria-hidden={true}
              className="fab fa-facebook"
            />
          </a>
          <a
            href="https://www.instagram.com/___mph___/"
            target="_blank"
          >
            <styled.i
              aria-hidden={true}
              className="fab fa-instagram"
            />
          </a>
          <a
            href="mailto:misha.medvedev2001@gmail.com"
            target="_blank"
          >
            <styled.i
              aria-hidden={true}
              className="fas fa-envelope-open-text"
            />
          </a>
        </styled.div>
      </styled.div>
      <styled.div>
        <h3>
          Source
        </h3>
        <a
          href="https://github.com/MedvedMichael/mph-web-site"
          target="_blank"
        >
          <styled.i
            aria-hidden={true}
            className="fab fa-github"
          />
        </a>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.footer>
`;
